pkg/framework/test/apiserver.go

```diff
@@ -53,6 +65,11 @@ func (s *APIServer) Start() error {
 		return err
 	}
 
+	err := s.Etcd.Start()
```

Comment:
I've been wondering about that for a while ... Should we expect the etcd process to be started or not? While it doesn't hurt to start if not started, I'm more curious about the parallel with stop: should we stop the etcd server when we stop the apiserver?

Reply:
I think we've previously chatted about a couple of different potential down-the-line use-cases that drive this behaviour out. I'm now not convinced I understood everything we chatted about, so I'd like to repeat my understanding here to double-check. One key potential use case was the case in which the user wants to use an aggregate APIServer. This aggregate APIServer might be composed of a number of SubAPIServers, each of which needs to communicate with Etcd. At that time we believed we were comparing two possible architectures, which I will call "the old way" and "the new way".

In "the old way", every APIServer (and hence in future, presumably every SubAPIServer) contained a reference to some configuration which pointed to the address of an Etcd. This is enough to talk to an Etcd but not to start it. This meant that the starting Etcd was the job of the Fixtures struct. One major disadvantage of this approach was that if each potential SubAPIServer needed its own independent Etcd, this could not be arranged. The Fixtures struct only knows how to start one of them.

In "the new way", every APIServer (and hence in future, presumably every SubAPIServer) contains an actual Etcd struct. This means that it has the configuration needed to connect to the Etcd once it's running, and also that it has the means to start and stop it. In this world, starting and stopping Etcd becomes the job of the APIServer, not the Fixtures struct. This makes it possible for potential future SubAPIServers to manage their own state, without interfering with each other, and crucially without interference from the Fixtures struct. If you want to share a single Etcd between all your SubAPIServers, it can be the responsibility of the APIServerAggregator to start and stop Etcd. If you want different Etcds for different SubAPIServers, then each SubAPIServer can manage its own.

Were we talking at crossed-purposes when we last discussed this? Is it worth chatting about it again?